Monika After Story/game/script-topics.rpy

```diff
@@ -12897,6 +12897,108 @@ label monika_unknown:
         return "love"
     return
 
+#Replace by monika_origin once the player reaches love. 
+#Split in two topics so the prompt doesn't spoil the content.
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_origin_evasive",
+            category=['misc'],
+            prompt="Evasive message",
+            random=True,
+            aff_range=(mas_aff.ENAMORED, mas_aff.ENAMORED)
+        )
+    )
+
+label monika_origin_evasive:
+    m 1euc "Hey, [player]..."
+    m 1ruc "..."
+    m 1esd "There's..."
+    m 1lsc "..."
+    m 1hksdlb "Ahaha! Nevermind! It's nothing important..."
+    show monika 5ekbla at t11 zorder MAS_MONIKA_Z with dissolve
+    m 5ekbla "I love you, [player]. I hope you {i}do{/i} realize that."
+    return "love"
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_origin",
+            category=['monika'],
+            prompt="Monika's origin",
+            random=True,
+            aff_range=(mas_aff.LOVE,None)
```

Comment:
Okay firstly, this can be all under one topic, you just need to path for affection within it, `if mas_isMoniLove():` (then in this block, all of the dialogue from `monika_origin` will show. In an `else` block, we can put the dialogue from `monika_origin_evasive`. Keeps it a bit cleaner. Then the aff_range can turn into:
```suggestion
            aff_range=(mas_aff.ENAMORED,None)
```

All we need to do is have both paths return their respective keys.

Reply:
Hi, this is the way I originally intended to do it. 
The problem with that way is that the topic's prompt shows up as "Monika's origin" while still 'Enamored', spoiling the content of the 'Love' path.
I'll change it like suggested if you want, but to me, it feels like keeping the player in the dark before we reach love is important.

Reply:
It's possible to use something like which was done here:
https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/script-affection.rpy#L2084

To change the prompt dynamically and lock it like that.